docs/cheatsheet.rst

```diff
@@ -168,3 +168,42 @@ Modifiers
 - ``override``: States that this function, modifier or public state variable changes
   the behavior of a function or modifier in a base contract.
 
+Optimizer Step Dependencies
+===========================
+
+=========================================== ===================================================== =============================================================== ===========================================
+Step                                        Hard prerequisites                                    Steps that should run before                                    Steps that should run after
+=========================================== ===================================================== =============================================================== ===========================================
+:ref:`a <ssa-transform>`                    Disambiguator, ForLoopInitRewriter                    ExpressionSplitter, CommonSubexpressionEliminator               UnusedAssignEliminator
+:ref:`C <conditional-simplifier>`           Disambiguator                                         SSATransform, DeadCodeEliminator
+:ref:`c <common-subexpression-eliminator>`  Disambiguator, ForLoopInitRewriter                    ExpressionSplitter, SSATransform                                UnusedPruner, UnusedAssignEliminator
```

Comment:
Initially I wanted to just add the extra info to the table we already have in the optimizer docs, but our current CSS only allows very narrow tables. I even had to cut full names here to avoid the scrollbar hiding most of the info.

The best compromise I found was to create a new table in the cheatsheet. I don't like it but I like other options much less.

The nice thing is that here I can sort the steps alphabetically by their abbreviations, which is much more useful than the original order when decoding the sequences.

Reply:
Now you have the large comment exactly on the first line I wanted to comment on :-D...
Why the ``SSATransform`` here? There *could* be different implementations for the data flow analyzer, which might mean that pseudo-SSA form helps - but currently, if anything, it may get more slower due to larger numbers of variables, won't it?

Reply:
I added your remarks to CSE description. I still left SSA in the table, but added a footnote explaining that it does not have to run before the step every time.